### ISSUE TITLE
Icons are required to meet the 3:1 contrast ratio(WCAG 1.4.11)

### ISSUE THREAD

**Issue by NeerajChoukseyDeque** [w3c/wcag#2152]
Hi Team,
 Do we need to check the 3:1 contrast ratio for the Avatar icons, please confirm?
I am not sure whether these icons are required to understand the content for low-vision users or [not.](url)

Also in the case of Emojis, do we need to check the 3:1 contrast ratio, please share your view.
I feel that since emojis convey emotional cues which can be considered important, they should meet the 3:1 contrast ratio.
![Avatar](https://user-images.githubusercontent.com/95408440/144388312-ea1c4ff2-1f83-4552-86aa-084fc01f126b.png)

Attaching the screenshot for reference.

**Comment by patrickhlauke** [comment]
if a user can't fully distinguish between the different avatars, will they have trouble understanding the content/purpose of the site? is it critical for them to be able to distinguish them fully? if yes, then in that case it's required. if no, then it's not.

**Comment by NeerajChoukseyDeque** [comment]
> if a user can't fully distinguish between the different avatars, will they have trouble understanding the content/purpose of the site? is it critical for them to be able to distinguish them fully? if yes, then in that case it's required. if no, then it's not.

I personally feel that the avatar is really not essential for the user to understand the purpose of the site or block low-vision users to perform the action.

**Comment by JAWS-test** [comment]
If the avatar is used in the application as the only means to identify a person, then the contrasts are important (e.g. to identify which person's comment is which).

If the avatar is only additional and the person's name is visible in text form, then I consider the contrasts to be unimportant.

**Comment by NeerajChoukseyDeque** [comment]
> If the avatar is used in the application as the only means to identify a person, then the contrasts are important (e.g. to identify which person's comment is which).
> 
> If the avatar is only additional and the person's name is visible in text form (as here), then I consider the contrasts to be unimportant.

Thank you. in my scenario, Avtar is just used to select the profile pic. What is the case with emojis?
![Emojis](https://user-images.githubusercontent.com/95408440/144419983-26facb56-4fb9-40af-b348-4ca3b51dd824.png)

**Comment by patrickhlauke** [comment]
if not being able to properly recognise/distinguish emojis means that instead of, say, a thumbs-up you end up posting a middle finger emoji...then yes, it'll be important to make sure at least the main/most distinguishing parts of the emoji have sufficient contrast.

anyway, i *think* this issue can be closed now, as the original question has been answered I feel (tl;dr: yes, 3:1 contrast is required if it's important that the icons/images/emojis can be recognised and distinguished)

**Comment by NeerajChoukseyDeque** [comment]
> Adding text could help. I find a lot of emoji that have text on focus and hover to meet this.

Thank you.
Thank you to everyone who contributed to it!!

**Comment by JAWS-test** [comment]
@mraccess77 

> Adding text could help. I find a lot of emoji that have text on focus and hover to meet this.

I'm not sure if this is sufficient according to WCAG, because it means that I have to mouse over all the icons or focus them with the keyboard to be able to recognize them. I may not even see the icon due to poor contrast and not know that there is something there that I can mouse over to get some text to pop up. Of course, such a text in the tooltip is better than nothing, but I would still consider it a violation of WCAG contrast requirements.

**Comment by JAWS-test** [comment]
@NeerajChoukseyDeque Can you please make the title meaningful, e.g. "Contrast of Avatars and Emojis"

**Comment by NeerajChoukseyDeque** [comment]
> @NeerajChoukseyDeque Can you please make the title meaningful, e.g. "Contrast of Avatars and Emojis"

Did the changes!!

**Comment by mraccess77** [comment]
I would like an official decision on whether information available on focus and hover would conform or not.  This is very common on charts as a way to meet this requirement.  We need to be consistent whether we allow or fail this.

**Comment by patrickhlauke** [comment]
feels like this is now a separate discussion ... "if something has low contrast by default (be it text or images), but then has acceptable contrast on focus/hover, does it fail text and/or non-text contrast requirements?"

my answer would be yes, it does fail. you can note that once it's focused/hovered, it has good contrast, but in its initial state it fails (and it can get to the point where if it has low contrast to start with, a user may not even be able to discern the thing and hover over it, for instance...)

**Comment by mraccess77** [comment]
What if the focusable hoverable trigger had good contrast - would that change your stance?  Say the trigger had good contrast but you had to focus the trigger to see the text for the chart item where the colors didn't have sufficient adjacent contrast.

We already allow today a table that can be displayed through a disclosure right?  Surely activating a link or disclosure triangle to display a data table would pass based on our current guidance.  Do you disagree with that?

I'm not trying to be confrontational - but trying to get at where the line is based on traditional acceptable practices under WCAG 2.

**Comment by JAWS-test** [comment]
Maybe open a new issue for the new discussion, especially since this one is already closed?
